app/models/manageiq/providers/ansible_tower/configuration_manager/refresh_parser.rb

```diff
@@ -68,9 +68,17 @@ def parse_host(host)
           :virtual_instance_ref => host.instance_id,
         }
 
+        cross_link_host(new_result)
+
         return uid, new_result
       end
 
+      def cross_link_host(new_result)
+        vm = Vm.where(:uid_ems => new_result[:virtual_instance_ref]).first
```

Comment:
I would have expected the ems_ref (or perhaps a ems_id, ems_ref pair) as opposed to the uid_ems.